File: __iOS_BACKUP__/ShapeByte/features/workout/schedule/data/WorkoutScheduleRepository.swift
//
//  WorkoutScheduleRepository.swift
//  ShapeByte
//
//  Created by Lang, Stefan [Shape Byte Tech] on 02.08.24.
//

import Foundation

class WorkoutScheduleRepository {
    private let datasource: WorkoutScheduleDatasource

    init(datasource: WorkoutScheduleDatasource) {
        self.datasource = datasource
    }

    func currentScheduleEntry() -> WorkoutScheduleEntry? {
        return datasource.currentWorkoutScheduleEntry()
    }
}

File: features/workout/data/contract/src/commonMain/kotlin/de/stefan/lang/shapebyte/features/workout/data/contract/schedule/WorkoutScheduleRepository.kt
package de.stefan.lang.shapebyte.features.workout.data.contract.schedule

import de.stefan.lang.foundation.core.contract.loadstate.LoadState

// TODO: Test
public class WorkoutScheduleRepository public constructor(
    private val datasource: WorkoutScheduleDatasource,
) {
    public suspend fun currentWorkoutScheduleEntry(): LoadState.Result<WorkoutScheduleEntry?> =
        datasource.currentWorkoutScheduleEntry()
}
